### PR TITLE
add throbber on task page

### PR DESCRIPTION
**Pull request by philbrookes**

### CODE REVIEW

**Comment by philbrookes** [comment]
@aidenkeating @david-martin @JameelB can you take a look at this, please?

File: src/pages/tutorial/task/task.js [hunk 1]
```diff
@@ -103,6 +104,15 @@ class TaskPage extends React.Component {
     setProgress(progress);
   };
 
+  areDocsAttributesReady = (attrs) => {
```
Comment (aidenkeating) [inline src/pages/tutorial/task/task.js]: I think we could make this walkthrough-independent by just ensuring the values of every attribute that's provided is not `null/undefined`?

Also, where did you get the gif from? It's awesome!

**Comment by pb82** [comment]
@philbrookes Looks like throbber.css is missing: `Module not found: Error: Can't resolve './throbber.css' in '/home/travis/build/integr8ly/tutorial-web-app/src/components/throbber'`

**Comment by mindreeper2420** [comment]
Instead of using an animated PNG, can we use the default PatternFly loading state? https://www.patternfly.org/pattern-library/communication/loading-state/

The icon can be manipulated using CSS animations, which are lighter on browser rendering.

**Comment by mindreeper2420** [comment]
There is a PatternFly React component already available for the spinner: http://patternfly-react.surge.sh/patternfly-3/index.html?knob-Loading=true&knob-Size=md&selectedKind=patternfly-react%2FWidgets%2FSpinner&selectedStory=Spinner&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

File: src/pages/tutorial/task/task.js (hunk 1 above)
Comment (philbrookes) [reply src/pages/tutorial/task/task.js]: I made it myself on loading.io :) I tried that initially, but some of the values are always null in some of the walkthroughs.

**Comment by philbrookes** [comment]
@pb82 thanks for pointing out the missing file, I've added it now.

**Comment by philbrookes** [comment]
@mindreeper2420 thanks for the heads-up, I'll take a look at converting this to use that.

**Comment by philbrookes** [comment]
@mindreeper2420 converted to the patternfly component now.

**Comment by mindreeper2420** [comment]
@philbrookes looks good! Would it be possible place the spinner on a white background, instead of the semi-transparent grey one? Similar to what is shown in the design documentation? Also, can the icon be increased to the large size? Thanks!

**Comment by mindreeper2420** [comment]
Just for reference, this is the Trello card for adding loading states: https://trello.com/c/kB7Mk18b/232-loading-state-for-a-walkthrough-while-namespace-is-being-set-up.

**Comment by philbrookes** [comment]
@mindreeper2420 @bdellasc Thanks guys, do we have some resources available for this? Such as the repeating animation and the backdrop image? Or should I be creating those too?

File: src/pages/tutorial/task/task.js (hunk 1 above)
Comment (aidenkeating) [reply src/pages/tutorial/task/task.js]: @philbrookes That issue you're seeing should end up being resolved by https://github.com/integr8ly/tutorial-web-app/pull/103 , basically some variables are currently only populated on the landing page so it causes that issue.

We can leave it for now anyway.

I think these two PRs will end up quite similar in terms of styling and stuff. There might be a possibility of using the same component at some stage in the future. I'll create a card.

**Comment by mindreeper2420** [comment]
@philbrookes can you access the Google Drive link that Brian shared? That has all of the resources in it. If not, I can try to provide another way.

**Comment by mindreeper2420** [comment]
@philbrookes for the CSS around it, feel free to look at the Congratulations page stylesheet. If you have any issues with it, feel free to let me know and I can help out.

**Comment by mindreeper2420** [comment]
@philbrookes we're using the SVG and not an animated gif for the animated boxes. You'll want to use the image `StartingServices_Final.svg`.

File: src/components/loadingScreen/loadingScreen.css [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+.loadingscreen-throbber {
+  display: inline-block;
+  position: fixed;
+  top: 0;
+  bottom: 50%;
+  left: 0;
+  right: 0;
+  margin: auto;
+  z-index: 501;
```
Comment (mindreeper2420) [inline src/components/loadingScreen/loadingScreen.css]: Why a `z-index` so high?

File: src/components/loadingScreen/loadingScreen.css [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.loadingscreen-text {
+  display: inline-block;
+  position: fixed;
+  color: white;
```
Comment (mindreeper2420) [inline src/components/loadingScreen/loadingScreen.css]: This should be a PatternFly color variable from `colors.scss`

File: src/components/loadingScreen/loadingScreen.css [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  position: fixed;
+  top: 45%;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  width: 50%;
+  margin: auto;
+  z-index: 501;
+  background-color: transparent !important;
+  border: 2px solid white;
```
Comment (mindreeper2420) [inline src/components/loadingScreen/loadingScreen.css]: Color variable

File: src/components/loadingScreen/loadingScreen.css [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  display: inline-block;
+  position: fixed;
+  top: 45%;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  width: 50%;
+  margin: auto;
+  z-index: 501;
+  background-color: transparent !important;
```
Comment (mindreeper2420) [inline src/components/loadingScreen/loadingScreen.css]: I believe this use of `!important` will be able to removed once added as a `sass` file. CSS specificity should take over and apply the transparency correctly.

File: src/components/loadingScreen/loadingScreen.css [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  border: 2px solid white;
+  color: white !important;
+}
+
+.loadingscreen-progressbar .progress-bar {
+  background-color: white;
```
Comment (mindreeper2420) [inline src/components/loadingScreen/loadingScreen.css]: Use PatternFly color variable.

**Comment by philbrookes** [comment]
@mindreeper2420 I believe all these issues are addressed now, if you wouldn't mind taking another look.

**Comment by mindreeper2420** [comment]
@philbrookes I can't get past the initial loading screen anymore to get into a task - are there any tricks to do so?

**Comment by philbrookes** [comment]
hey @mindreeper2420 are you sure you've pulled the latest edition of this branch? I had left some development code in which kept that page displayed while I restyle it, however I have taken that out now.

**Comment by mindreeper2420** [comment]
@philbrookes there appears to be a bug with OpenShift not loading the environment, which is why I can't get into the webapp. The code is good, so I'm going to mark this as approved  👍 .

**Pull request merged**